src/lib/comms/sol-lwm2m-bs-server.c

```diff
@@ -465,10 +470,15 @@ sol_lwm2m_bootstrap_server_write(struct sol_lwm2m_bootstrap_server *server,
     enum sol_coap_response_code response_code),
     const void *data)
 {
+    enum sol_lwm2m_path_props props;
+
     SOL_NULL_CHECK(server, -EINVAL);
     SOL_NULL_CHECK(client, -EINVAL);
     SOL_NULL_CHECK(path, -EINVAL);
 
+    props = sol_lwm2m_common_get_path_props(path);
+    SOL_EXP_CHECK(props == PATH_IS_INVALID, -EINVAL);
```

Comment:
This check should be `path < PATH_HAS_INSTANCE`.
The separate API `sol_lwm2m_bootstrap_server_write_object()` must be used if `path == PATH_HAS_OBJECT`.